pkg/controller/autoscaler/horizontalpodautoscaler_controller.go

```diff
@@ -113,7 +141,89 @@ func (a *HorizontalPodAutoscalerController) reconcileAutoscalers() error {
 			continue
 		}
 
+		var metrics heapster.MetricResultList
+		err = json.Unmarshal(resultRaw, &metrics)
+		if err != nil {
+			glog.Warningf("Failed to unmarshall heapster response: %v", err)
+			continue
+		}
+
 		glog.Infof("Metrics available for %s: %s", reference, string(resultRaw))
+
+		currentConsumption, count := metricSpec.aggregator(metrics)
+		if count != len(podList.Items) {
+			glog.Warningf("Metrics obtained for %d/%d of pods", count, len(podList.Items))
+			continue
+		}
+
+		// if the ratio is 1.2 we want to have 2 replicas
+		desiredReplicas := 1 + int((currentConsumption.Quantity.MilliValue()*int64(count))/hpa.Spec.Target.Quantity.MilliValue())
```

Comment:
The formula there is slightly broken anyway:

`argetNumOfPods = sum(CurrentPodsConsumption) / Target`

If we have target 0.5 and current consumption is 0.95 then the floor of the above is still 1.0 but the cpu is burning. Moreover with just one cpu on node you will never get to the point when the service is scaled.